conventions.md

```diff
@@ -0,0 +1,68 @@
+# Starlark Language Conventions
+
+This document describes non-normative but generally recommended conventions for
+Starlark.
+
+## API Documentation
+
+### Docstrings
+
+API documentation for a function should be provided in a [Python-style
+docstring](https://peps.python.org/pep-0257/) - a string literal which is the
+first line of the function's body.
+
+By convention, this string should consist of an optional newline, followed by
+a 1-line, 1-sentence summary, optionally followed by a blank line and blank-line
+delimited paragraphs of additional documentation text.
+
+If a string literal is the first statement in a Starlark source file, it is
+treated as the documentation for that file; the conventions are the same as for
+a function's docstring.
+
+For example:
+
+```python
+"""A collection of useful utilities"""
+
+def is_valid(config):
+    """
+    Verifies if `config` is a valid configuration.
+
+    Usage example:
+
+        is_valid({"example": {"cpu": "arm"}})  # returns True
+    """
+    ...
+```
+
+Documentation processors should take care to dedent common leading whitespace
+from a multiline docstring's lines (note that the first line could have no
+leading whitespace).
+
+### Doc comments
+
+API documentation for a constant may be provided in
```

Comment:
Not a "constant" (which I'd take to mean a lexical construct) -- a global variable.

But we can go further and say it's for any assignment statement. Or we can add on that it should only be for the first assignment statement, in the case of multiple assignments to the same local var. (Not sure what should happen if you violated that and tried to have two docs for the same local var.) Or we can leave it as is and have the implementation behavior of attaching doc strings to local assignments be an undocumented behavior.

Reply:
Changed to global. Starlark doesn't allow reassigning a global, so the multiple assignment concern is moot :)